app/services/katello/repository_type_manager.rb

```diff
@@ -79,6 +79,19 @@ def indexable_content_types
                   flatten
       end
 
+      def generic_content_types(enabled_only = true)
+        repo_types = enabled_only ? enabled_repository_types : defined_repository_types
+        list = repo_types.values.map do |type|
+          type.content_types.select { |ct| ct.model_class::CONTENT_TYPE == Katello::GenericContentUnit::CONTENT_TYPE }
+        end
+        list.flatten
+      end
+
+      def generic_content_type?(content_type, enabled_only = true)
+        types = generic_content_types(enabled_only)
+        !types.select { |ct| ct.content_type == content_type }.empty?
```

Comment:
I think you could just do `types.include?(content_type)` if you changed `generic_content_types` to return a list of content type labels.